### PR TITLE
feat: Retrive recent pull requests for GithubToolkit

### PR DESCRIPTION
**Pull request by rsrbk**
## Description

Additional function to GithubToolkit to retrieve list of pull requests.

## Motivation and Context

A function to retrieve recent pull requests will be helpful for things like building a summary for open sourced repo contributions to post on social media.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/github_toolkit.py [hunk 1]
```diff
@@ -71,6 +72,63 @@ def summary(self) -> str:
         )
 
 
+@dataclass
+class GithubPullRequestDiff:
```
Comment (zechengz) [inline camel/toolkits/github_toolkit.py:76]: nit:
```suggestion
class GitHubPullRequestDiff:
```

File: camel/toolkits/github_toolkit.py [hunk 2]
```diff
@@ -71,6 +72,63 @@ def summary(self) -> str:
...
+    r"""Represents a single diff of a pull request on Github."""
+
+    def __init__(self, filename: str, patch: str) -> None:
+        r"""Initialize a GithubPullRequestDiff object.
```
Comment (zechengz) [inline camel/toolkits/github_toolkit.py:80]: This docstring is not necessary

File: camel/toolkits/github_toolkit.py [hunk 3]
```diff
@@ -71,6 +72,63 @@ def summary(self) -> str:
...
+
+        Args:
```
Comment (zechengz) [inline camel/toolkits/github_toolkit.py:82]: You can put these args docstring to the class one.

File: camel/toolkits/github_toolkit.py [hunk 4]
```diff
@@ -71,6 +72,63 @@ def summary(self) -> str:
...
+            filename (str): The name of the file that was changed.
+            patch (str): The diff patch for the file.
+        """
+        self.filename = filename
+        self.patch = patch
+
+    def summary(self) -> str:
```
Comment (zechengz) [inline camel/toolkits/github_toolkit.py:89]: Maybe you can override `__str__(self):` and you can just use str(GithubPullRequestDiff) to get the summary

File: camel/toolkits/github_toolkit.py [hunk 5]
```diff
@@ -71,6 +72,63 @@ def summary(self) -> str:
...
+        Args:
+            title (str): The title of the GitHub pull request.
+            body (str): The body/content of the GitHub pull request.
+            diffs (List[GithubPullRequestDiff]): A list of diffs for the pull request.
+        """
+        self.title = title
+        self.body = body
+        self.diffs = diffs
+
+    def summary(self) -> str:
```
Comment (zechengz) [inline camel/toolkits/github_toolkit.py:122]: Maybe here also override the __str__

File: examples/function_call/github_examples.py [hunk 6]
```diff
@@ -68,8 +114,8 @@ def solve_issue(
 
 
 def main(model=None) -> None:
-    repo_name = "camel-ai/test-github-agent"
-    solve_issue(repo_name=repo_name, issue_number=1, model=model)
+    repo_name = "camel-ai/camel"
```
Comment (zechengz) [inline examples/function_call/github_examples.py:117]: Can we add `argparse` to allow user pass the repo path?

File: camel/toolkits/github_toolkit.py (hunk 1 above)
Comment (rsrbk) [reply camel/toolkits/github_toolkit.py:76]: I'm not super sure about this one, cause I tried to follow the same naming style as https://github.com/PyGithub/PyGithub which we use underneath and they follow lowercase "h"

**Review by Wendong-Fan — APPROVED** [review]
Thanks @rsrbk ! Left some comment, overall LGTM

**Review by zechengz — APPROVED** [review]
LGTM after resolving @Wendong-Fan 's comments

**Pull request merged**